data-prepper-core/src/main/java/org/opensearch/dataprepper/peerforwarder/PeerForwarder.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.peerforwarder;
+
+import com.amazon.dataprepper.model.event.Event;
+import com.amazon.dataprepper.model.record.Record;
+import org.opensearch.dataprepper.peerforwarder.discovery.StaticPeerListProvider;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.linecorp.armeria.client.WebClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpMethod;
+import com.linecorp.armeria.common.HttpRequest;
+import com.linecorp.armeria.common.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.net.NetworkInterface;
+import java.net.SocketException;
+import java.net.UnknownHostException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class PeerForwarder {
+    private static final Logger LOG = LoggerFactory.getLogger(PeerForwarder.class);
+    private static final WebClient LOCAL_CLIENT = null;
+    public static final int ASYNC_REQUEST_THREAD_COUNT = 200;
+
+    private final PeerForwarderClientFactory peerForwarderClientFactory;
+    private final ObjectMapper objectMapper;
+    private HashRing hashRing;
+    private PeerClientPool peerClientPool;
+    private final ExecutorService executorService;
+
+    public PeerForwarder(final PeerForwarderClientFactory peerForwarderClientFactory,
+                         final ObjectMapper objectMapper) {
+        this.peerForwarderClientFactory = peerForwarderClientFactory;
+        this.objectMapper = objectMapper;
+
+        executorService = Executors.newFixedThreadPool(ASYNC_REQUEST_THREAD_COUNT);
+    }
+
+    public List<Record<Event>> processRecords(final Collection<Record<Event>> records, final Set<String> identificationKeys) {
+        // TODO: decide the default behaviour of Core Peer Forwarder and move the HashRing and PeerClientPool creation to constructor
```

Comment:
QUES: what default behavior needs to be decided?

Reply:
I discussed about this with @dlvenable offline. The default behavior of core peer forwarder is that it's enabled by default, but the default `PeerListProvider` is `StaticPeerListProvider` which has a check in it's constructor if the static list is empty or not. By default it'll be empty and throws a `RuntimeException`. So, I moved the creation of `HashRing` and `PeerClientPool` to method. This will be injected in to constructor once we decide the default behavior.

Reply:
I am interested in this going forward and looking forward to the update. For a static provider, creating this in the constructor makes sense. I want to make sure we are considering the hashRing may need to be updated for dynamic peer providers like cloudMap and DNS